docfx_yaml/extension.py

```diff
@@ -1002,6 +1031,43 @@ def pretty_package_name(package_group):
     return " ".join(capitalized_name)
 
 
+# Given generated markdown files, incorporate them into the docfx_yaml output.
+# The markdown file metadata will be added to top level of the TOC.
+def find_markdown_pages(app, outdir):
+    # Use this to ignore markdown files that are unnecessary.
+    files_to_ignore = [
+        "index.md",     # merge index.md and README.md and index.yaml later
+        "reference.md", # Reference docs overlap with Overview. Will try and incorporate this in later.
+        "readme.md",    # README does not seem to work in cloud site
+        "upgrading.md", # Currently the formatting breaks, will need to come back to it.
+    ]
+
+    markdown_dir = Path(app.builder.outdir).parent / "markdown"
+    if not markdown_dir.exists():
+        print("There's no markdown file to move.")
+        return
+
+    # For each file, if it is a markdown file move to the top level pages.
+    for mdfile in markdown_dir.iterdir():
+        if mdfile.is_file() and mdfile.name.lower() not in files_to_ignore:
+            shutil.copy(mdfile, f"{outdir}/{mdfile.name.lower()}")
+
+            # Extract the header name for TOC.
+            with open(mdfile) as f:
+                header_line = f.readline()
+                # Ignore licenses and other non-headers prior to the header.
+                while "#" not in header_line:
```

Comment:
Check prefix instead of just `'#'` in the line in case it's somewhere else in the line, possibly escaped?

Reply:
I've seen few cases where they do end up in the middle of the file after they're processed by the markdown builder for Sphinx. I don't think I'd have to worry about it being present escaped after it's been processed by `sphinx-build`. Though this is something good to consider if I want to move markdown files directly from the source rather than using the processed versions.